text/0000-Bundle-Load-API.md

```diff
@@ -51,18 +52,26 @@ The output of the API will be the function pointer type supplied by the API user
 * 
 * @return A function pointer symbols of type T supplied by the user or nullptr if the library is not loaded
 * 
-* @throws std::runtime_error if this symbol fetched is nullptr
+* @throws std::runtime_error if the bundle is not started and active
+* @throws std::runtime_error if the symbol fetched is nullptr
```

Comment:
For a bundle to be active means that the shared library has been loaded into the process. The way the return value and throwing guarantees are worded it sounds like this function can never return ```nullptr```. 

One question I have is, what's the purpose of throwing an exception to the user of this API? What are the pros and cons of:
1. not throwing at all and returning ```nullptr``` for any kind of failure.
2. throwing for any failure case (i.e. bundle not active, symbol doesn't exist) and only returning a valid symbol pointer
3. throwing if the bundle is not started and returning ''nullptr'' if the bundle is active and the symbol doesn't exist or a valid symbol pointer if the bundle is active and the symbol exists.

Reply:
@jeffdiclemente  I checked the use case of DS and got overly inspired from it in terms of exceptions thrown from this API. I think its best if we not throw at all i.e. go with option 1) since
Pros:
1. puts the onus of throwing exceptions on clients (e.g. DS)
2. allows us to write cleaner code with just one branch in it

Cons:
1. Clients will get less info as to what went wrong (e.g. bundle not active or no symbol found or OS API failed some arcane error, though I am not sure if clients would love to know the last one)

2nd option might entail clients to check for exception(s) in each failure case scenario. Its something this that current design is doing but is worded incorrectly

3rd option provides balance of both : returning nullptr if symbol not available and throwing an exception when bundle is not started. It will be similar to other APIs which check for bundle validity and throw if that's not the case.

Let me know if you think 1) is better than 3) or vice versa.

Reply:
I think that option 1's con is a significant one. When trying to debug a complex system, this type of error messaging is useful. I would not want to lose information. I'm leaning towards option 3.